scopes/pipelines/builder/builder.graphql.ts

```diff
@@ -1,36 +1,123 @@
+  path: string;
+  content?: string;
+  downloadUrl?: string;
+};
+type ArtifactGQLData = ArtifactObject & { files: ArtifactGQLFile[]; componentId: ComponentID };
+type TaskReport = PipelineReport & {
+  artifact: ArtifactGQLData;
+  componentId: ComponentID;
+};
+const FILE_PATH_PARAM_DELIM = '~';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export function builderSchema(builder: BuilderMain) {
+export function builderSchema(builder: BuilderMain, scope: Scope) {
   return {
     typeDefs: gql`
-      type ExtensionDescriptor {
-        # extension ID.
-        id: String
+      type TaskReport {
+        id: String!
+        name: String!
+        description: String
+        startTime: String
+        endTime: String
+        errors: [String!]
+        warnings: [String!]
+        artifact(path: String): Artifact
+      }
 
-        # icon of the extension.
-        icon: String
+      type ArtifactFile {
+        # for GQL caching - same as the path
+        id: String!
+        name: String
+        path: String!
+        # artifact file content (only for text files). Use /api/<component-id>/~aspect/builder/<extension-id>/~<path> to fetch binary file data
+        content: String
+        downloadUrl: String
       }
 
-      type ExtensionArtifact {
-        # descriptor of the artifact's extension
-        extensionDescriptor: ExtensionDescriptor
+      type Artifact {
+        # artifact name
+        name: String
+        description: String
+        files: [ArtifactFile!]!
+      }
+
+      type AspectData {
+        # aspectId
+        id: String!
+        data: JSONObject
       }
 
       extend type Component {
-        # list of extension artifacts.
-        getArtifacts(hash: String!): [ExtensionArtifact]
+        pipelineReport(taskId: String): [TaskReport!]!
       }
     `,
+
     resolvers: {
       Component: {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        getArtifacts: async (component: Component, { hash }: { hash: string }) => {
-          // const artifacts = await builder.getArtifacts(component.id, hash);
-          // return artifacts.map((artifact) => artifact.toObject());
+        pipelineReport: async (component: Component, { taskId }: { taskId?: string }) => {
+          const builderData = builder.getBuilderData(component);
+
+          const artifactsByTask = (builderData?.artifacts || [])
+            ?.filter((artifact) => !taskId || artifact.task.id === taskId)
+            .reduce((accum, next) => {
+              accum.set(next.task.id, next);
+              return accum;
+            }, new Map<string, ArtifactObject>());
+
+          const pipelineReport = (builderData?.pipeline || [])
+            .filter((pipeline) => !taskId || pipeline.taskId === taskId)
+            .map((pipeline) => ({
+              ...pipeline,
+              artifact: artifactsByTask?.get(pipeline.taskId),
+              componentId: component.id,
+            }));
```

Comment:
even if you don't do the previous comment, I believe you can use one of the APIs mentioned above to learn these filters here